src/Tizen.NUI.Components/Controls/ItemsView/GridLayouter.cs

```diff
@@ -0,0 +1,405 @@
+            int division = pureIndex / spanSize;
+            int remainder = pureIndex % spanSize;
+            int emptyArea = isHorizontal ? (int)(colView.Size.Height - (sizeCandidate.Height * spanSize)) :
+                                            (int)(colView.Size.Width - (sizeCandidate.Width * spanSize));
+            if (division < 0) division = 0;
+            if (remainder < 0) remainder = 0;
+
+            xPos = isHorizontal ? division * sizeCandidate.Width + (hasHeader ? headerSize : 0) : emptyArea * align + remainder * sizeCandidate.Width;
+            yPos = isHorizontal ? emptyArea * align + remainder * sizeCandidate.Height : division * sizeCandidate.Height + (hasHeader ? headerSize : 0);
+
+            return new Position(xPos, yPos);
+        }
+
+        private void FindVisibleItems(Vector2 visibleArea)
+        {
+            bool hasHeader = (colView.Header != null);
+            bool hasFooter = (colView.Footer != null);
+            int MaxIndex = colView.InternalItemSource.Count - 1 - (hasFooter ? 1 : 0);
+
+            // Header is Showing
+            if (hasHeader && visibleArea.X < headerSize)
+            {
+                FirstVisible = 0;
+
+            }
+            else
+            {
+                float visibleAreaX = visibleArea.X - (hasHeader ? headerSize : 0);
+                FirstVisible = (Convert.ToInt32(Math.Abs(visibleAreaX / StepCandidate)) - 1) * spanSize;
+
+                if (FirstVisible < 0) FirstVisible = 0;
+                if (hasHeader) FirstVisible += 1;
+            }
+            
+            if (hasFooter && visibleArea.Y > ScrollContentSize - footerSize)
+            {
+                LastVisible = MaxIndex + 1;
+            }
+            else
+            {
+                float visibleAreaY = visibleArea.Y - (hasHeader ? headerSize : 0);
+                //Need to Consider GroupHeight!!!!
+                LastVisible = (Convert.ToInt32(Math.Abs(visibleAreaY / StepCandidate)) + 1) * spanSize;
+
+                if (hasHeader) LastVisible += 1;
+                if (LastVisible > (MaxIndex)) LastVisible = MaxIndex;
+            }
+        }
+
+        private ViewItem GetVisibleItem(int index)
+        {
+            foreach (ViewItem item in VisibleItems)
+            {
+                if (item.Index == index) return item;
+            }
+
+            return null;
+        }
+
+        /// <summary>
+        /// Clean up ItemsLayouter.
+        /// </summary>
+        /// <param name="view"> ItemsView of layouter. </param>
+        /// <since_tizen> 8 </since_tizen>
+        /// This may be public opened in tizen_6.0 after ACR done. Before ACR, need to be hidden as inhouse API
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public override void Initialize(ItemsView view)
+        {
+            colView = view as CollectionView;
+            if (colView == null)
+            {
+                throw new NotSupportedException();
```

Comment:
`ArgumentException`이 적절해 보입니다.
```suggestion
                throw new ArgumentException(["error message"], nameof(view));
```